### DISCUSSION TITLE
first capture taking long

### DISCUSSION THREAD

**Discussion by sygolene** [jim-easterbrook/python-gphoto2#177] [General]
Hello Jim, 
I've noticed that the first capture_photo would take significantly longer than subsequent ones.
The first capture will complete in 7-10 seconds while a normal capture takes about 1.2s with my current gear.
Once this "slow" one is done, everything performs fine. This behaviour cannot be reproduced while operating the camera alone so I would exclude the camera.

An exemple of timestamps doing a simple 5 shots loop of file_path = camera.capture(gp.GP_CAPTURE_IMAGE) without wait_for_events.
Apart from trying to sneak in a "dummy" shot to get over it, is there anything I could do?

Thanks in advance, 

Starting standard capture loops
2024-09-09 17:44:22.981 before capture
Capturing image
2024-09-09 17:44:30.991 - after capture
2024-09-09 17:44:30.992 before capture
Capturing image
2024-09-09 17:44:32.263 - after capture
2024-09-09 17:44:32.264 before capture
Capturing image
2024-09-09 17:44:33.427 - after capture
2024-09-09 17:44:33.427 before capture
Capturing image
2024-09-09 17:44:34.679 - after capture
2024-09-09 17:44:34.679 before capture
Capturing image
2024-09-09 17:44:35.855 - after capture

**Comment by jim-easterbrook** [comment]
Try setting manual focus and manual exposure on the camera.

Does the gphoto2 command line program suffer similar delays?

There is nothing in python-gphoto2 that could create this delay, so I believe it's a problem in libgphoto2.

**Reply by sygolene** [reply]
I was already in manual everything and just gave it a try in an interval gphoto2 cmd... Indeed that's where it lies. 

I'll try and reach their maintainers. 

Thank you

**Reply by sygolene** [reply]
Hello Jim, 

In case you're interested, a little follow-up, file system size related, it seemed. 

https://github.com/gphoto/libgphoto2/issues/1018

cheers,

**Reply by jim-easterbrook** [reply]
Ah, that's something worth remembering.
